### PR TITLE
Add schema.org markup for emails

### LINKED ISSUE #1371: Schema.org for emails
Google provides rich snippets that may be used in emails:
https://developers.google.com/gmail/markup/overview

Potential usage:
1. [Order confirmation email](https://developers.google.com/gmail/markup/reference/order)
2. [Invoice sent to user](https://developers.google.com/gmail/markup/reference/invoice)
3. [Confirm/reset password action](https://developers.google.com/gmail/markup/reference/one-click-action)

**Acceptance criteria**

- Implemented as JSON-LD
- Generated on the backend, passed to the template
- All changes are documented under `seo` at Read the Docs
- [Need for registering with Google](https://developers.google.com/gmail/markup/registering-with-google) in order to make it work on production servers is mentioned

**Todo**
1. Implement Schema.org for suitable mails in our project
1. Copy email sample and test using [Markup Data Tester](https://www.google.com/webmasters/markup-tester/)
1. Copy email sample and test using [Apps Script](https://developers.google.com/gmail/markup/apps-script-tutorial)

**Helpful resources**
[Testing schema for emails](https://developers.google.com/gmail/markup/testing-your-schema)

### PR DESCRIPTION
**Pull request by Pacu2**
I've added schema markup for Order Confirmation Email.
Apart from that, did a little clean up in emails.py module and its tests.

## Action button on Desktop:
action_button_gmail

## Confirmation card on the mobile:
image uploaded from ios

Close #1371 
- [x] Docs 
- [x] Tests
- [x] Functionality
- [x] Validated with Google tools

I want to merge this change because...

(Please mention all relevant issue numbers.)

(If your changes affect the UI, providing "before" and "after" screenshots will
greatly reduce the amount of work needed to review your work.)

### Pull Request Checklist

(Please keep this section. It will make maintainer's life easier.)

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

File: saleor/order/emails_helpers.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+def get_product_data(line, currency, organization):
+    product_data = {
+        '@type': 'Offer',
+        'itemOffered': {
+            '@type': 'Product',
+            'name': line.product_name,
+            'sku': line.product_sku,
+        },
+        'price': line.unit_price_gross * line.quantity,
```
Comment (mad-anne) [inline saleor/order/emails_helpers.py]: `OrderLine` has method `get_total()` that could be used here.

File: tests/test_emails_helpers.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import pytest
+
+from django.contrib.sites.models import Site
+
+from saleor.order.emails_helpers import (
+    get_order_confirmation_markup, get_organization, get_product_data)
+
+
+def test_get_organization(site_settings):
```
Comment (akjanik) [inline tests/test_emails_helpers.py]: Is it necessary to use this fixture if `Site.objects.get_current()` is called and modified below anyway? I think we can remove fixture OR use `site_settings.site.name = example_name`.
It is also to consider using Mock here, although mocking `name` can be tricky.

Comment (Pacu2) [reply tests/test_emails_helpers.py]: I'd rather not mock anything unless completely necessary. Removed unused fixture

File: tests/test_emails_helpers.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    site.save()
+
+    result = get_organization()
+    assert result['name'] == example_name
+
+
+def test_get_product_data_without_image(order_with_lines):
+    """Tested OrderLine Product has no image assigned."""
+    order_line = order_with_lines.get_lines()[0]
+    organization = get_organization()
```
Comment (akjanik) [inline tests/test_emails_helpers.py]: I would rather not use in test a function that is not being a subject of the test. Considering it being called also in `test_get_product_data_with_image`, I think we may create fixture for that.

File: tests/test_emails_helpers.py (hunk 2 above)
Comment (akjanik) [reply tests/test_emails_helpers.py]: Actually, after reading rest of the tests, I think we should use the fixture and use assignment `site = site_settings.site`, similar as we do in other tests - that's why we have this fixture. Calling model unrelated to the test that is actually referring to the `seetings.py` file, see:
https://github.com/django/django/blob/master/django/contrib/sites/models.py#L50
Our fixtures already uses pytest's `settings` fixture, so we should be safe here.
Moreover, call to `Sites.objects.get_current` is being used in more than one test.

File: tests/conftest.py [hunk 4]
```diff
@@ -71,8 +71,15 @@ def request_cart_with_item(product_in_stock, request_cart):
 
 
 @pytest.fixture
-def order(billing_address):
-    return Order.objects.create(billing_address=billing_address)
+def order(billing_address, customer_user):
+    return Order.objects.create(
+        billing_address=billing_address,
+        user_email=customer_user.email,
+        total_net=Money(100, currency='USD'),
```
Comment (rafalp) [inline tests/conftest.py]: Lets drop `currency=` form `Money()`, its remnant from back when we had `Price()` where it was third arg and we already dropped that in Saleor's codebase.

File: saleor/order/emails.py [hunk 5]
```diff
@@ -5,52 +5,78 @@
...
+    # Order confirmation template requires additional information
+    if template == CONFIRM_ORDER_TEMPLATE:
+        email_markup = get_order_confirmation_markup(order)
+        email_context.update({'order': order, 'schema_markup': email_markup})
+    return {
+        'recipient_email': recipient_email, 'template': template,
+        'context': email_context}
+
+
+def _send_confirmation(recipient_email, template, context):
```
Comment (elwoodxblues) [inline saleor/order/emails.py]: What's the benefit of using this function? Can't we just use `send_templated_email` directly in `send_order_confirmation` and other functions that send emails below?

File: saleor/order/emails.py [hunk 6]
```diff
@@ -5,52 +5,78 @@
...
     order = Order.objects.get(pk=order_pk)
-    email = order.get_user_current_email()
-    url = build_absolute_uri(
-        reverse('order:details', kwargs={'token': order.token}))
-    return {'email': email, 'url': url, 'template': template}
+    recipient_email = order.get_user_current_email()
+    email_context = get_email_context(order.token)
+
+    # Order confirmation template requires additional information
+    if template == CONFIRM_ORDER_TEMPLATE:
```
Comment (elwoodxblues) [inline saleor/order/emails.py]: This logic is specific only to `send_order_confirmation`. Shouldn't it live there?

File: saleor/order/emails.py [hunk 7]
```diff
@@ -5,52 +5,78 @@
 from templated_email import send_templated_mail
 
 from ..core.utils import build_absolute_uri
+from .emails_helpers import get_order_confirmation_markup
```
Comment (elwoodxblues) [inline saleor/order/emails.py]: `emails_helpers` contains only functions related to the "email markup" feature, so maybe a better name would be `email_markup.py` plus a module docstring explaining that it's related to the Google's email markup feature? 
On the other hand, we have functions that generate `schema.org` markup in few other places, maybe we should group them into one app?

File: saleor/order/emails.py (hunk 6 above)
Comment (Pacu2) [reply saleor/order/emails.py]: I need `Order` to create the markup, didn't want to fetch it twice

File: saleor/order/emails.py (hunk 7 above)
Comment (Pacu2) [reply saleor/order/emails.py]: Grouped as /schema/ app

**Pull request merged**